src/test/java/Homework19_20.java

```diff
@@ -0,0 +1,43 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class Homework19_20 extends BaseTest{
+    @Test
+
+    public void deletePlaylist () throws InterruptedException {
```

Comment:
Please, delete `throws InterruptedException`

By removing the throws InterruptedException, you are indicating that the method does not expect to encounter InterruptedException and is not using any blocking or sleeping operations that would lead to such an exception. 

For example

**Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.**

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.

![image](https://github.com/eolsen3/A47-QA-Automation-Framework/assets/104761482/65c243bd-b12e-4e3c-b3ee-3e04c16036d5)

```
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.testng.Assert;
import org.testng.annotations.Test;

public class Homework19_20 extends BaseTest {

    @Test
    public void deletePlaylist() {
        String deletedPlayListMsg = "Deleted Playlist";

        provideEmail("erika.olsen@testpro.io");
        providePassword("te$t$tudent");
        clickSubmit();
        openPlaylist();
        clickDeletePlaylistBtn();

        String actualMsg = getDeletedPlaylistMsg();
        String expectedMsg = "Deleted playlist \"Test Pro Playlist.\"";
        Assert.assertEquals(actualMsg, expectedMsg);
    }

    public void openPlaylist() {
        WebElement clickTestProPlaylist = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(".playlist:nth-child(3)")));
        clickTestProPlaylist.click();
    }

    public void clickDeletePlaylistBtn() {
        WebElement deletePlaylist = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[title='Delete this playlist']")));
        deletePlaylist.click();
    }

    public String getDeletedPlaylistMsg() {
        WebElement successMessage = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("div.success.show")));
        return successMessage.getText();
    }
}
```